include/flang/ISO_Fortran_binding.h

```diff
@@ -139,6 +125,28 @@ typedef struct CFI_cdesc_t {
   CFI_dim_t dim[CFI_ISO_FORTRAN_BINDING_FLEXIBLE_ARRAY]; /* must appear last */
 } CFI_cdesc_t;
 
+/* 18.5.4 */
+#ifdef __cplusplus
+// C++ does not support zero-sized array (rank=0)
+// Also, CFI_cdesc_t already contains 1 dim in cpp
+template<bool B, int rank> struct Cond_CFI_cdesc_t {
```

Comment:
Please make sure that these declarations are hidden in a namespace.

Reply:
If you really need a flexible sized object (with size >= 0) that's co-allocated in juxtaposition in C++, I'd suggest perhaps a class-specific member placement new and some accessor members.

Reply:
Thanks placement new is an interesting idea. I am not sure I can use it for what I want here though. I just want to create a type (not object) that is the correct size and alignment to hold a `CFI_cdesc_t` for an element of rank `rank` . The standard does not really care of what this type is made of. From what I understand of placement new, it would more fit to the next step to be done by the user:
`CFI_CDESC_T(rank) storage;
CFI_cdesc_t * dv = new (&storage) CFI_cdesc_t;`
Maybe it is possible to optimize the space by using member placement new inside the CFI_cdesc_t class in C++ but the standard is really precise on what CFI_cdesc_t structure should be so I do not want to tweak it too much in C++.
For now, I think CFI_CDESC_T is good enough in C++.